### PR TITLE
Added prow job configurations for unit-tests, integration-tests, dev and test image builds

### PR DESCRIPTION
**Pull request by unmarshall**
**What this PR does / why we need it**:
Adds prow job configurations for unit tests, integration tests, dev & test image builds

**Special notes for your reviewer**:

### CODE REVIEW

File: config/jobs/etcd-druid/etcd-druid-build-dev-images.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          - name: image-builder
+            image: europe-docker.pkg.dev/gardener-project/releases/ci-infra/image-builder:v20241111-d15d824
+            command:
+              - /image-builder
+            args:
+              - --log-level=info
+              - --docker-config-secret=gardener-prow-gcr-docker-config
+              - --registry=europe-docker.pkg.dev/gardener-project/releases/ci-infra
+              - --target=etcd-druid
+              - --context=hack/tools/image
```
Comment (oliver-goetz) [inline config/jobs/etcd-druid/etcd-druid-build-dev-images.yaml:29]: I'm not sure what you are trying to build here 😅

Comment (oliver-goetz) [reply config/jobs/etcd-druid/etcd-druid-build-dev-images.yaml:29]: I found a concourse job which builds dev images on HEAD updates. Thus, this whole jobs might not be needed, does it?

Comment (unmarshall) [reply config/jobs/etcd-druid/etcd-druid-build-dev-images.yaml]: Thanks for providing the info over chat. I have made the corrections for etcd-druid and also for dwd

**Comment by oliver-goetz** [comment]
Thanks 🚀 

/lgtm
/approve

**Pull request merged**